src/Features/Core/Portable/Diagnostics/AnalyzerHelper.cs

```diff
@@ -307,6 +308,32 @@ private static void AssertCompilation(Project project, Compilation compilation1)
             Contract.ThrowIfFalse(compilation1 == compilation2);
         }
 
+        /// <summary>
+        /// Return true if the given <paramref name="analyzer"/> is suppressed for the given project.
+        /// NOTE: This API is intended to be used only for performance optimization.
+        /// </summary>
+        public static bool IsAnalyzerSuppressedForProject(DiagnosticAnalyzer analyzer, Project project)
+        {
+            var options = project.CompilationOptions;
+            if (options == null || analyzer == FileContentLoadAnalyzer.Instance || analyzer.IsCompilerAnalyzer())
```

Comment:
what's teh FileContentLoadAnalayzer?  Also, doc this condition.  e.g. why these two are never suppressed.

Reply:
instead of hardcoding this in, could these analyzers implement/expose something to indicate this info?

Reply:
`FileContentLoadAnalayzer` is a special analyzer that Tomas implemented a while back - it reports special IDE-only warnings when we are unable to loading text for a document fails. This analyzer is hardcoded at bunch of places in the IDE, so any change to remove hardcoding with a special property should be done as a separate cleanup PR.